llvm/include/llvm/Analysis/TargetTransformInfoImpl.h

```diff
@@ -65,7 +65,8 @@ class TargetTransformInfoImplBase {
     (void)PSI;
     (void)BFI;
     JTSize = 0;
-    return SI.getNumCases();
+    bool HasDefault = !SI.defaultDestUndefined();
+    return SI.getNumCases() + HasDefault;
```

Comment:
Seems like this was just a bug not counting the default to begin with. Maybe can submit this separately, since any regression probably comes from this?

Reply:
> Seems like this was just a bug not counting the default to begin with. Maybe can submit this separately, since any regression probably comes from this?

Could you explain further? Do you mean putting the default branch processing into `InlineCost`(Make code in InlineCost better understood?)?

Reply:
I mean getNumCases never included the cost of the default branch. Now it does, if it's not only an unreachable inst. I initially read this as trying to exclude the cost of the default branch